src/main/java/org/openpnp/machine/reference/driver/TinygDriverZPneumatic.java

```diff
@@ -0,0 +1,502 @@
+/*
+ 	Copyright (C) 2011 Jason von Nieda <jason@vonnieda.org>
+ 	
+ 	This file is part of OpenPnP.
+ 	
+	OpenPnP is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    OpenPnP is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with OpenPnP.  If not, see <http://www.gnu.org/licenses/>.
+ 	
+ 	For more information about OpenPnP visit http://openpnp.org
+ */
+
+package org.openpnp.machine.reference.driver;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.concurrent.TimeoutException;
+
+import javax.swing.Action;
+
+import org.openpnp.gui.support.PropertySheetWizardAdapter;
+import org.openpnp.gui.support.Wizard;
+import org.openpnp.machine.reference.ReferenceActuator;
+import org.openpnp.machine.reference.ReferenceHead;
+import org.openpnp.machine.reference.ReferenceHeadMountable;
+import org.openpnp.machine.reference.ReferenceNozzle;
+import org.openpnp.machine.reference.driver.wizards.AbstractSerialPortDriverConfigurationWizard;
+import org.openpnp.model.LengthUnit;
+import org.openpnp.model.Location;
+import org.openpnp.spi.PropertySheetHolder;
+import org.simpleframework.xml.Attribute;
+import org.simpleframework.xml.Element;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * TODO: Consider adding some type of heartbeat to the firmware.
+ * TODO: The whole movement wait lock thing has to go. See if we can do a
+ * P4 type command like the other drivers to wait for movement to complete.
+ * Disabled axes don't send status reports, so movement wait lock never happens.
+ * Probably short moves also won't.
+ */
+public class TinygDriverZPneumatic extends AbstractSerialPortDriver implements Runnable {
```

Comment:
Good Idea Jason !
Thank you for your observations @cri-s <https://github.com/cri-s> !
Sorry for my mistakes. I will fix them and I will come back.

On Mon, Feb 29, 2016 at 7:21 PM, Jason von Nieda <notifications@github.com>
wrote:

> In
> src/main/java/org/openpnp/machine/reference/driver/TinygDriverZPneumatic.java
> <https://github.com/openpnp/openpnp/pull/237#discussion_r54443842>:
>
> > +import org.simpleframework.xml.Element;
> > +import org.slf4j.Logger;
> > +import org.slf4j.LoggerFactory;
> > +
> > +import com.google.gson.JsonObject;
> > +import com.google.gson.JsonParser;
> > +import com.google.gson.JsonSyntaxException;
> > +
> > +/**
> > + * TODO: Consider adding some type of heartbeat to the firmware.
> > + * TODO: The whole movement wait lock thing has to go. See if we can do a
> > + * P4 type command like the other drivers to wait for movement to complete.
> > + * Disabled axes don't send status reports, so movement wait lock never happens.
> > + * Probably short moves also won't.
> > + */
> > +public class TinygDriverZPneumatic extends AbstractSerialPortDriver implements Runnable {
>
> It looks like this class is mostly a copy of TinygDriver with changes to
> moveTo. Could it just extend TinygDriver instead of duplicating all the
> code?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openpnp/openpnp/pull/237/files#r54443842>.
>